src/join-5-auth.sol

```diff
@@ -34,49 +32,58 @@ interface GemLike {
 
 // Authed GemJoin for a token that has a lower precision than 18 and it has decimals (like USDC)
 
-contract AuthGemJoin5 is LibNote {
+contract AuthGemJoin5 {
     // --- Auth ---
     mapping (address => uint256) public wards;
-    function rely(address usr) external note auth { wards[usr] = 1; }
-    function deny(address usr) external note auth { wards[usr] = 0; }
+    function rely(address usr) external auth { wards[usr] = 1; emit Rely(usr); }
+    function deny(address usr) external auth { wards[usr] = 0; emit Deny(usr); }
     modifier auth { require(wards[msg.sender] == 1); _; }
 
-    VatLike public vat;
-    bytes32 public ilk;
-    GemLike public gem;
-    uint256 public dec;
+    VatLike public immutable vat;
+    bytes32 public immutable ilk;
+    GemLike public immutable gem;
+    uint256 public immutable dec;
     uint256 public live;  // Access Flag
 
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event Join(address indexed urn, uint256 wad, address indexed msgSender);
+    event Exit(address indexed guy, uint256 wad);
+
     constructor(address vat_, bytes32 ilk_, address gem_) public {
         gem = GemLike(gem_);
-        dec = gem.decimals();
-        require(dec < 18, "GemJoin5/decimals-18-or-higher");
+        uint256 dec_ = dec = GemLike(gem_).decimals();
+        require(dec_ < 18, "GemJoin5/decimals-18-or-higher");
         wards[msg.sender] = 1;
+        emit Rely(msg.sender);
         live = 1;
         vat = VatLike(vat_);
         ilk = ilk_;
     }
 
-    function cage() external note auth {
```

Comment:
we need an event emission here if we're removing note

Reply:
This should be added to here as well then: https://github.com/makerdao/dss-gem-joins/blob/master/src/join-managed.sol

Reply:
Done.

Reply:
New issue https://github.com/makerdao/dss-gem-joins/issues/23